Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/repository/storage/VaccinationContainer.kt

```diff
@@ -1,48 +1,43 @@
 package de.rki.coronawarnapp.vaccination.core.repository.storage
 
 import androidx.annotation.Keep
-import com.google.gson.Gson
 import com.google.gson.annotations.SerializedName
 import de.rki.coronawarnapp.ui.Country
 import de.rki.coronawarnapp.vaccination.core.VaccinatedPersonIdentifier
 import de.rki.coronawarnapp.vaccination.core.VaccinationCertificate
-import de.rki.coronawarnapp.vaccination.core.common.RawCOSEObject
+import de.rki.coronawarnapp.vaccination.core.certificate.RawCOSEObject
+import de.rki.coronawarnapp.vaccination.core.certificate.VaccinationDGCV1
 import de.rki.coronawarnapp.vaccination.core.personIdentifier
-import de.rki.coronawarnapp.vaccination.core.qrcode.HealthCertificateCOSEDecoder
 import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateCOSEParser
 import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateData
 import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateQRCode
-import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateV1
-import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateV1Parser
 import de.rki.coronawarnapp.vaccination.core.server.valueset.VaccinationValueSet
 import org.joda.time.Instant
 import org.joda.time.LocalDate
 
 @Keep
-data class VaccinationContainer(
+data class VaccinationContainer internal constructor(
     @SerializedName("vaccinationCertificateCOSE") val vaccinationCertificateCOSE: RawCOSEObject,
     @SerializedName("scannedAt") val scannedAt: Instant,
 ) {
 
+    // Either set by [ContainerPostProcessor] or via [toVaccinationContainer]
+    @Transient lateinit var parser: VaccinationCertificateCOSEParser
```

Comment:
this looks strange to me. Parser inside a data class ?! , can we improve this somehow?

Reply:
I'm not 100% happy either, but so far the alternative seem  also not nice.
Would be another wrapper around this class and/or more logic related to this piece of data in a parent class.
If you have a good idea maybe we can have a call later.